crowbar_framework/app/models/node_object.rb

```diff
@@ -1064,6 +1064,30 @@ def ssh_cmd(cmd)
     [200, {}]
   end
 
+  # Check for the presence of given file on the node
+  def file_exists?(file)
+    out = run_ssh_cmd("test -e #{file}")
+    out[:exit_code].zero?
+  end
+
+  # Waits until the background script finishes.
+  # We expect that the script generates two kinds of files to indicate success or failure
+  # Raises timeout exception if the waiting time exceedes 'seconds'
+  def wait_for_script_to_finish(seconds, ok_file, failed_file)
```

Comment:
@aspiers What about this? https://gist.github.com/jsuchome/5a76ab1851e8c7934cd0c1a681f1a01e (diff)
https://gist.github.com/jsuchome/1e20c4a306bce77477014f6fe6a4a59b (new code)